core/src/main/kotlin/keiyoushi/utils/Json.kt

```diff
@@ -19,10 +22,20 @@ inline fun <reified T> String.parseAs(json: Json = jsonInstance): T =
  * Parses the response body into an object of type [T].
  */
 inline fun <reified T> Response.parseAs(json: Json = jsonInstance): T =
-    use { json.decodeFromStream(body.byteStream()) }
+    json.decodeFromResponse(serializer(), this)
 
 /**
  * Serializes the object to a JSON string.
  */
 inline fun <reified T> T.toJsonString(json: Json = jsonInstance): String =
     json.encodeToString(this)
+
+@PublishedApi
+internal fun <T> Json.decodeFromResponse(
+    deserializer: DeserializationStrategy<T>,
+    response: Response,
+): T {
+    return response.body.byteStream().use {
+        decodeFromStream(deserializer, it)
+    }
```

Comment:
And we can use `decodeFromBufferedSource()` like https://github.com/mihonapp/mihon/commit/2d19729869d1c88a46c0d1a4582d29a80d7ab9cb if the dependency on `kotlinx-serialization-json-okio` is ensured.